lib/tasks/pubmed.rake

```diff
@@ -3,6 +3,76 @@ namespace :pubmed do
     @client ||= PubmedClient.new
   end
 
+  desc 'update pubmed source records and rebuild pub_hashes for publications for a given list of cap_profile_ids passed in a text file'
+  # filename should be a list of cap_profile_id, one per line with no header
+  # call with RAILS_ENV=production bundle exec rake pubmed:refresh_source_records_for_cap_profile_ids['filename.txt']
+  task :refresh_source_records_for_cap_profile_ids, [:filename] => :environment do |_t, args|
+    filename = args[:filename]
+    fail "filename is required." unless filename.present?
+    cap_profile_ids=File.readlines(filename)
```

Comment:
I'd modify the call stack on this rake task.  First create a rake task that accepts just one PMID.  Then create a rake task that accepts one CAP profile id (it can then call the PMID task).  Then create rake task that takes a file of profile-ids.  The break-down of the rake tasks will help to reduce the complexity of this one and provide more flexibility at the rake command line.